### PR TITLE
std.container.Array - documentation cleanup

### PR DESCRIPTION
**Pull request by byebye**
- make formatting consistent
- add missing details, e.g. about exception being thrown
- fix wrong descriptions
- fix wrong complexity specifications
- delete redundant comments

### CODE REVIEW

File: std/container/array.d [hunk 1]
```diff
@@ -265,14 +265,13 @@ if (!is(Unqual!T == bool))
         size_t _capacity;
         T[] _payload;
 
-        // Convenience constructor
```
Comment (quickfur) [inline std/container/array.d]: Why is the comment deleted?

File: std/container/array.d [hunk 2]
```diff
@@ -389,49 +365,50 @@ if (!is(Unqual!T == bool))
...
                 reserve(1 + capacity * 3 / 2);
             }
             assert(capacity > length && _payload.ptr);
-            emplace(_payload.ptr + _payload.length, stuff);
+            emplace(_payload.ptr + _payload.length, elem);
             _payload = _payload.ptr[0 .. _payload.length + 1];
             return 1;
         }
 
-        /// Insert a range of items
```
Comment (quickfur) [inline std/container/array.d]: Why is this comment un-ddoc'd?

File: std/container/array.d (hunk 1 above)
Comment (byebye) [inline std/container/array.d]: It is a private structure and at least for me these kind of comments - where comment is basically the same as function name - are completely redundant.

File: std/container/array.d (hunk 2 above)
Comment (byebye) [inline std/container/array.d]: As above.

**Comment by byebye** [comment]
Could you point me the undocumented methods? It seems `insertBack` is documented... or maybe you mean the `struct Payload` which is private?

File: std/container/array.d (hunk 1 above)
Comment (quickfur) [inline std/container/array.d]: Ah, makes sense. Thanks!

File: std/container/array.d (hunk 2 above)
Comment (quickfur) [inline std/container/array.d]: Got it.

**Comment by quickfur** [comment]
Gah, the way github shows the diffs makes it easy to miss that fold that hid the "private struct Payload" declaration.  I should have looked at it locally from a git fetch instead. Sorry for these unfounded nitpicks. Your changes look good. Let's give it the go.

**Comment by byebye** [comment]
Great, thanks!

File: std/container/array.d [hunk 3]
```diff
@@ -2111,23 +2112,22 @@ if (is(Unqual!T == bool))
             _store._backend.length = _store._backend.length - 1;
         }
     }
+
     /// ditto
     alias stableRemoveBack = removeBack;
```
Comment (quickfur) [inline std/container/array.d]: Why is this alias duplicated below (line 2145)?

Comment (quickfur) [inline std/container/array.d]: Oh... these are overloads... I see.  But are two aliases necessary?

Comment (byebye) [inline std/container/array.d]: Uhm.. I don't think so, didn't notice that one :) Removed.

Comment (byebye) [inline std/container/array.d]: Wait, I think it is because of documentation - this way both aliases are documented accordingly. So it should stay.

Comment (quickfur) [inline std/container/array.d]: Fair enough.

**Pull request merged**